File: Leetcode/cpp/sol3174.cpp

//  https://leetcode.com/problems/clear-digits/description/

class Solution {
public:
    string clearDigits(string s) {

        string rslt = "";
        for (char i:s){
            if (!isdigit(i)){
                rslt+=i;
            }else{
                if(!rslt.empty()){
                    rslt.pop_back();
                }
            }
        }
        return rslt;
    }
};

File: Leetcode/python/sol3174.py
#  https://leetcode.com/problems/clear-digits/
class Solution:
    def clearDigits(self, s: str) -> str:

        stack = []
        for i, j in enumerate(s, start=1):
            if j.isalpha():
                stack.append(j)
            else:
                stack.pop()

        return "".join(stack)

